exist-core/src/main/java/org/exist/xquery/functions/system/Restore.java

```diff
@@ -52,23 +52,40 @@
 
 	protected final static Logger logger = LogManager.getLogger(Restore.class);
 
-	public final static FunctionSignature signature = new FunctionSignature(
-			new QName("restore", SystemModule.NAMESPACE_URI, SystemModule.PREFIX),
-			"Restore the database or a section of the database (admin user only).",
-			new SequenceType[] {
-					new FunctionParameterSequenceType("dir-or-file", Type.STRING, Cardinality.EXACTLY_ONE,
-							"This is either a backup directory with the backup descriptor (__contents__.xml) or a backup ZIP file."),
-					new FunctionParameterSequenceType("admin-pass", Type.STRING, Cardinality.ZERO_OR_ONE,
-							"The password for the admin user"),
-					new FunctionParameterSequenceType("new-admin-pass", Type.STRING, Cardinality.ZERO_OR_ONE,
-							"Set the admin password to this new password.") 
-			}, 
-			new FunctionReturnSequenceType(Type.NODE, Cardinality.EXACTLY_ONE, "the restore results"));
+	public final static FunctionSignature signatures[] = {
+	        new FunctionSignature(
+                    new QName("restore", SystemModule.NAMESPACE_URI, SystemModule.PREFIX),
+                    "Restore the database or a section of the database (admin user only).",
+                    new SequenceType[] {
+                            new FunctionParameterSequenceType("dir-or-file", Type.STRING, Cardinality.EXACTLY_ONE,
+                                    "This is either a backup directory with the backup descriptor (__contents__.xml) or a backup ZIP file."),
+                            new FunctionParameterSequenceType("admin-pass", Type.STRING, Cardinality.ZERO_OR_ONE,
+                                    "The password for the admin user"),
+                            new FunctionParameterSequenceType("new-admin-pass", Type.STRING, Cardinality.ZERO_OR_ONE,
+                                    "Set the admin password to this new password.")
+                    },
+                    new FunctionReturnSequenceType(Type.NODE, Cardinality.EXACTLY_ONE, "the restore results")),
```

Comment:
Could reduce the copy-paste here by extracting common params into static final variables... or use the FunctionDSL.